pkg/middleware/authentication.go

```diff
@@ -0,0 +1,29 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/gazebo-web/auth/pkg/authentication"
+	"net/http"
+	"strings"
+)
+
+// BearerToken is an HTTP middleware that provides a token to any authentication.Authentication implementation.
+// The token is extracted from the HTTP header called Authorization, including the prefix "Bearer ".
+func BearerToken(auth authentication.Authentication) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			h := r.Header.Get("Authorization")
+			hh := strings.Split(h, "Bearer ")
+			token := hh[1]
```

Comment:
We should handle the case where there is no header authorization header at all and return an error if that is the case.